libr/bin/p/bin_elf.c

```diff
@@ -1034,6 +1034,29 @@ static bool has_canary(RBinFile *bf) {
 	return ret;
 }
 
+static int has_retguard(RBinFile *bf) {
```

Comment:
i would go for bool instead of int. -1 case is not really giving too much informtion in here

Reply:
Intentional here because it s relevant only for OpenBSD here thus rabin2 does not display this info for other oses.

Reply:
To explain further, openbsd/clang can compile with -fret-protector adding thus a retguard protection for each return function into .openbsd.randomdata section here we try to indicate only in the case of OpenBSD if this binary had been compiled with this flag or not.

Reply:
@devnexen you forgot the case when OpenBSD binary can be opened on the other platform.

Reply:
Ah, nevermind, ignore my comment, you mean the case of OpenBSD binary.